lib/trino-hive-formats/src/main/java/io/trino/hive/formats/rcfile/RcFileReader.java

```diff
@@ -144,100 +143,107 @@ private RcFileReader(
         this.end = offset + length;
         verify(end <= fileSize, "offset plus length is greater than data size");
 
-        // read header
-        Slice magic = input.readSlice(RCFILE_MAGIC.length());
-        boolean compressed;
-        if (RCFILE_MAGIC.equals(magic)) {
-            version = input.readByte();
-            verify(version <= CURRENT_VERSION, "RCFile version %s not supported: %s", version, inputFile.location());
-            validateWrite(validation -> validation.getVersion() == version, "Unexpected file version");
-            compressed = input.readBoolean();
-        }
-        else if (SEQUENCE_FILE_MAGIC.equals(magic)) {
-            validateWrite(validation -> false, "Expected file to start with RCFile magic");
+        this.input = new TrinoDataInputStream(inputFile.newStream());
+        try {
```

Comment:
Makes sense to me.

Given that the `input` stream is created as part of the constructor I don't see a way to verify this fix through unit testing.

Reply:
Looking at the code, author's intention was to call `corrupt` for all validations, and thus should close the stream.

I agree it's simpler to do this defensively with try-catch